launcher/src/main.rs

```diff
@@ -144,28 +164,72 @@ struct RawConfig {
     quantization_config: Option<QuantizationConfig>,
     n_embd: Option<usize>,
     hidden_size: Option<usize>,
+    intermediate_size: Option<usize>,
     num_attention_heads: Option<usize>,
+    num_key_value_heads: Option<usize>,
+    num_hidden_layers: Option<usize>,
     head_dim: Option<usize>,
     vision_config: Option<VisionConfig>,
     is_encoder_decoder: Option<bool>,
+    #[serde(rename = "num_experts_per_tok")]
+    experts: Option<usize>,
 }
 
 #[derive(Deserialize)]
 struct QuantizationConfig {
     quant_method: Option<Quantization>,
 }
 
-#[derive(Deserialize)]
+#[derive(Debug, Deserialize)]
 struct VisionConfig {}
 
-#[derive(Deserialize)]
+#[derive(Debug, Deserialize)]
 struct Config {
     max_position_embeddings: Option<usize>,
     quantize: Option<Quantization>,
     head_dim: Option<usize>,
+    num_heads: Option<usize>,
+    num_kv_heads: Option<usize>,
+    num_layers: Option<usize>,
+    intermediate_size: Option<usize>,
+    hidden_size: Option<usize>,
     model_type: Option<String>,
     vision_config: Option<VisionConfig>,
     is_encoder_decoder: bool,
+    experts: Option<usize>,
+}
+
+impl Config {
+    fn flop(&self) -> Option<u64> {
+        if self.vision_config.is_some() {
+            // VLM are much harder to predict and VRAM requirements
+            // Are more complex.
+            return None;
+        }
+        let num_heads = self.num_heads? as u64;
+        let num_kv_heads = self.num_kv_heads? as u64;
+        let head_dim = self.head_dim? as u64;
+        let hidden_size = self.hidden_size? as u64;
+        let intermediate_size = if let Some(experts) = self.experts {
+            (self.intermediate_size? * experts) as u64
+        } else {
+            self.intermediate_size? as u64
+        };
+        let num_layers = self.num_layers? as u64;
+
+        let q_flops = 2 * num_heads * head_dim * hidden_size;
+        let k_flops = 2 * num_kv_heads * head_dim * hidden_size;
+        let v_flops = 2 * num_kv_heads * head_dim * hidden_size;
+        let attn_flops = 2 * num_heads * head_dim * hidden_size;
+        let o_flops = 2 * num_heads * head_dim * hidden_size;
```

Comment:
No it should be OK. Since I use the flops on hardware in f16 already.

Both are divided by 2 for fp8 so the calculation should hold, it's just expressed in f16 unit.